rancher/distros-test-framework
# secrets encryption testcase

#### Proposed Changes ####


Adding test for secrets encryption workflows:

    traditional workflows: prepare, rotate, reencrypt
    New workflow: rotate-keys
    Note: rotate-keys feature is not ready yet for rke2.



Added documentation help as well.

#### Types of Changes ####


Patch Validation test
#### Testing ####


Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.

No

2. Verify code lint; we should not have errors.
Yes

3. Update the documentation if needed.
Yes

4. Update makefile and docker run if adding new tests.
Yes

5. Run your tests at least 4 times with all configurations needed and possible.
Yes

6. If needed test with different os types.
N/A

#### Linked Issues ####



#### Further Comments ####
